packages/dma-contracts/tasks/verify-deployment/index.ts

```diff
@@ -0,0 +1,279 @@
+import { ADDRESS_ZERO } from '@deploy-configurations/constants'
+import {
+  ConfigEntry,
+  SystemConfig,
+  SystemConfigEntry,
+} from '@deploy-configurations/types/deployment-config'
+import { Network } from '@deploy-configurations/types/network'
+import { ServiceRegistry } from '@typechain/index'
+import { color } from 'console-log-colors'
+import { task } from 'hardhat/config'
+
+import {
+  getServiceRegistry,
+  isInvalidAddress,
+  ServiceRegistryMaybe,
+  VerificationResult,
+} from '../common'
+const { red, yellow, green } = color
+
+function getValidationStatusString(verificationResult: VerificationResult): string {
+  const verificationPercentage: number = Math.round(
+    (verificationResult.totalValidated / verificationResult.totalEntries) * 100,
+  )
+
+  if (verificationPercentage < 30) {
+    return (
+      red(`${verificationPercentage.toFixed(2)}%`) +
+      ` (${verificationResult.totalValidated}/${verificationResult.totalEntries})`
+    )
+  } else if (verificationPercentage < 70) {
+    return (
+      yellow(`${verificationPercentage.toFixed(2)}%`) +
+      ` (${verificationResult.totalValidated}/${verificationResult.totalEntries})`
+    )
+  } else {
+    return (
+      green(`${verificationPercentage.toFixed(2)}%`) +
+      ` (${verificationResult.totalValidated}/${verificationResult.totalEntries})`
+    )
+  }
+}
+
+async function validateContracts(
+  config: { [key: string]: SystemConfigEntry },
+  serviceRegistry: ServiceRegistry,
+): Promise<VerificationResult> {
+  let totalEntries = 0
+  let totalValidated = 0
+
+  if (Object.keys(config).length === 0) {
+    return {
+      success: false,
+      totalEntries,
+      totalValidated,
+    }
+  }
+
+  for (const entryName in config) {
+    if (entryName === 'ServiceRegistry') {
+      continue
+    }
+
+    totalEntries++
+
+    const configEntry: SystemConfigEntry = config[entryName]
+    if (configEntry.serviceRegistryName) {
+      const registryAddress = await serviceRegistry.getRegisteredService(
+        configEntry.serviceRegistryName,
+      )
+
+      if (registryAddress === ADDRESS_ZERO) {
+        console.log(`${entryName}: ❌ (not deployed)`)
+        continue
+      }
+
+      if (registryAddress.toLowerCase() !== configEntry.address.toLowerCase()) {
+        console.log(
+          `${entryName}: ❌ (address mismatch ${configEntry.address} <--> ${registryAddress})`,
+        )
+        continue
+      }
+    }
+
+    totalValidated++
+
+    console.log(`${entryName}: ✅ (${configEntry.address})`)
+  }
+
+  return {
+    success: totalEntries === totalValidated,
+    totalEntries,
+    totalValidated,
+  }
+}
+
+async function validateDependencies(config: {
+  [key: string]: ConfigEntry
+}): Promise<VerificationResult> {
+  let totalEntries = 0
+  let totalValidated = 0
+
+  if (Object.keys(config).length === 0) {
+    return {
+      success: false,
+      totalEntries,
+      totalValidated,
+    }
+  }
+
+  for (const entryName in config) {
```

Comment:
what do you think about:
```
Object.entries(config).forEach(([key, value] ) => { })
```
Then, we won't need to create a `configEntry` and once again iterate through the `config` object.

Reply:
Agree! I wasn't sure what was more readable in TS